### PR TITLE
[Fully Modular] Adds breaching hammers

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Replaces unoriginal and unfunny baton with a far more interesting tool
Still got to add sprites and fix a runtime.  
Sprites added and runtimes fixed.
## Why It's Good For The Game
1.Lets security coordinate to breach doors , lets them cuck door spam.
2.Removes knockdown from batons , but provides a viable combat option when used in numbers , but not alone against CQC,  Martial arts , e-sword , etc.

On hit it applies 40 Stamina , and puts a 5 second cooldown for the security officer on any action. It also provides a passive block chance of 45% for melle and 15% for projectiles.1 Hit is enough to get someone in slowdown and keep them in range and force them to face the might of the thunk or fight against it with whatever they have.
## Changelog
:cl:
add: Security breaching hammers , that can break doors when used in pairs and act like shields + stunbatons in combat.
/:cl:

### CODE REVIEW

**Comment by dominuslucian** [comment]
good fucking meme

**Comment by softcerv** [comment]
I think that they could be neat as like a late round research locked weapon, but I feel that giving sec officers a tool to destroy doors as part of their standard kit could create a massive headache for engineering staff along with other potential issues.

**Comment by MLGTASTICa** [comment]
> I think that they could be neat as like a late round research locked weapon, but I feel that giving sec officers a tool to destroy doors as part of their standard kit could create a massive headache for engineering staff along with other potential issues.

Security is expected to only use these when needed, and well , if they needlessly break doors , the HOS can and should punish them , so can sergeants. It also needs 2 people to do it .

**Comment by softcerv** [comment]
> 
> 
> > I think that they could be neat as like a late round research locked weapon, but I feel that giving sec officers a tool to destroy doors as part of their standard kit could create a massive headache for engineering staff along with other potential issues.
> 
> Security is expected to only use these when needed, and well , if they needlessly break doors , the HOS can and should punish them , so can sergeants. It also needs 2 people to do it .

Letting sec bulldoze through departmental doors sounds incredibly painful for both antags and whoever is working in a department. Maybe the door breaching function could be treated the same way as lethals or it might require HoS/Captain permission.  Having some way to check what officers used the tool or another other form of accountability would help a lot.

**Comment by MLGTASTICa** [comment]
> > > I think that they could be neat as like a late round research locked weapon, but I feel that giving sec officers a tool to destroy doors as part of their standard kit could create a massive headache for engineering staff along with other potential issues.
> > 
> > 
> > Security is expected to only use these when needed, and well , if they needlessly break doors , the HOS can and should punish them , so can sergeants. It also needs 2 people to do it .
> 
> Letting sec bulldoze through departmental doors sounds incredibly painful for both antags and whoever is working in a department. Maybe the door breaching function could be treated the same way as lethals or it might require HoS/Captain permission. Having some way to check what officers used the tool or another other form of accountability would help a lot.

Witnesses exist.

**Comment by softcerv** [comment]
> 
> 
> > > > I think that they could be neat as like a late round research locked weapon, but I feel that giving sec officers a tool to destroy doors as part of their standard kit could create a massive headache for engineering staff along with other potential issues.
> > > 
> > > 
> > > Security is expected to only use these when needed, and well , if they needlessly break doors , the HOS can and should punish them , so can sergeants. It also needs 2 people to do it .
> > 
> > 
> > Letting sec bulldoze through departmental doors sounds incredibly painful for both antags and whoever is working in a department. Maybe the door breaching function could be treated the same way as lethals or it might require HoS/Captain permission. Having some way to check what officers used the tool or another other form of accountability would help a lot.
> 
> Witnesses exist.

Yes, they do, but they aren't always present or willing to talk. Even having something minor, like metal debris on the hammer to show that it was used to breach, would go a long way.

**Comment by MLGTASTICa** [comment]
> > > > > I think that they could be neat as like a late round research locked weapon, but I feel that giving sec officers a tool to destroy doors as part of their standard kit could create a massive headache for engineering staff along with other potential issues.
> > > > 
> > > > 
> > > > Security is expected to only use these when needed, and well , if they needlessly break doors , the HOS can and should punish them , so can sergeants. It also needs 2 people to do it .
> > > 
> > > 
> > > Letting sec bulldoze through departmental doors sounds incredibly painful for both antags and whoever is working in a department. Maybe the door breaching function could be treated the same way as lethals or it might require HoS/Captain permission. Having some way to check what officers used the tool or another other form of accountability would help a lot.
> > 
> > 
> > Witnesses exist.
> 
> Yes, they do, but they aren't always present or willing to talk. Even having something minor, like metal debris on the hammer to show that it was used to breach, would go a long way.

That a IC issue , im not going to add hard evidence for something that theres already plenty for. If you want this done do another PR that adds onto this if this gets merged.

**Comment by YakumoChen** [comment]
This makes it useless for security to apprehend people in melee. What are they supposed to do, spam flash people? Riddle them with rubber bullets? This creates far more problems than it solves (and fabricates a problem that doesn't need a solution)

blockchance on a standard issue weapon is incredibly cringe by the way, especially since it stacks with riot shields.
For reference the _esword_ has a 50 block chance in comparison.

**Comment by MLGTASTICa** [comment]
> This makes it useless for security to apprehend people in melee. What are they supposed to do, spam flash people? Riddle them with rubber bullets? This creates far more problems than it solves (and fabricates a problem that doesn't need a solution)
> 
> blockchance on a standard issue weapon is incredibly cringe by the way, especially since it stacks with riot shields.

Riot shields will break. Also yes you should be using rubbers, pepper , shoves and flashes to appherend people , not just spamming a baton and insta winning.

**Comment by YakumoChen** [comment]
Literally all of those things you just listed can be completely mitigated with a gas mask and sunglasses.

Shoving doesn't actually do anything on Skyrat other than annoy people.

**Comment by MLGTASTICa** [comment]
> Literally all of those things you just listed can be completely mitigated with a gas mask and sunglasses.
> 
> Shoving doesn't actually do anything on Skyrat other than annoy people.

Shoving hardstuns them for 6 seconds if they are on the ground. Also the hammer will stamcrit someone in 3 hits.

**Comment by YakumoChen** [comment]
I have literally never seen a shove stun work on Skyrat. Shove fights are endless.

15 seconds to hardstun a single person is frankly absurd. You're making the only option for security to even do their most basic job of arresting people for non-antag crimes an absurd chore or forcing them to just shoot everyone with rubbers (which will cause extreme problems with the crew)

**Comment by MLGTASTICa** [comment]
> I have literally never seen a shove stun work on Skyrat. Shove fights are endless.
> 
> 15 seconds to hardstun a single person is frankly absurd. You're making the only option for security to even do their most basic job of arresting people for non-antag crimes an absurd chore or forcing them to just shoot everyone with rubbers (which will cause extreme problems with the crew)

You get an absurb amount of spare ammo anyway , i don't really see an issue with that.

**Comment by YakumoChen** [comment]
This also more or less fucks over half the reason to use CQC, which specifically has a special interaction with security batons.

**Comment by MLGTASTICa** [comment]
> This also more or less fucks over half the reason to use CQC, which specifically has a special interaction with security batons.

CQC has its special interaction with all melee weapons , it also does the opposite , trying to solo a CQC user with a hammer is going to severely hamper the officer , unless the antag is very incapable.

**Comment by YakumoChen** [comment]
These damage values are insane. 15 force? _35_ throw force?

We only _just_ got rid of the forced brute damage on batons and now you want to replace them with something worse that forces "less than lethal security" again.

File: modular_skyrat/modules/sec_haul/code/peacekeeper/peacekeeper_hammer.dm [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	righthand_file = 'modular_skyrat/modules/sec_haul/icons/peacekeeper/baton/peacekeeper_baton_righthand.dmi'
+	slot_flags = ITEM_SLOT_BELT
+	force = 15
+	throwforce = 35
+	wound_bonus = 30
+	bare_wound_bonus = 40
+	block_chance = 45
+	w_class = WEIGHT_CLASS_NORMAL
+	attack_verb_continuous = list("whacks","breaches","bulldozes","flings","thwachs")
+	attack_verb_simple = list("breache","hammer","whack","slap","thwach","fling")
```
Comment (KathrinBailey) [inline modular_skyrat/modules/sec_haul/code/peacekeeper/peacekeeper_hammer.dm]:
```suggestion
	attack_verb_simple = list("breach","hammer","whack","slap","thwach","fling")
```

**Review by KathrinBailey — CHANGES_REQUESTED** [review]
1. Your branch is hecked. Check files changed.
2. I'm fine with this existing but I'd rather you keep this in the armoury and not given to every officer. Up to three in the armoury is fine.
3. Please post a video of the breaching or post a .gif of it. Kinda need to see how this performs in-game.
4. Lower block chance significantly and make bullets not be blocked at all, or just have 2% block chance or something.
5. Consider making this something you wield as a requirement to smash the doors, it'll also make having high stamina damage okay if you are required to wield the hammer to use it effectively. Wielded can have higher block chance for melee attacks, considering you're holding a bar in front of you with two hands, as opposed to limply holding a huge hammer with one hand.

Overall very good PR, just needs tweaking.

**Comment by Gandalf2k15** [comment]
Given that you've put effort into this, let's give it a TM, make sure you actually replace the batongs.

File: code/modules/unit_tests/rcd.dm [hunk 2]
```diff
@@ -43,4 +43,4 @@
 	for(var/obj/structure/frame/machine_frame in adjacent_turf.contents)
 		frame_count++
 
-	TEST_ASSERT_EQUAL(frame_count, 1, "Expected no density machine deconstruction frame stacking test to end up with exactly 1 machine frame.")
+	TEST_ASSERT_EQUAL(frame_count, 1, "Expected no density machine deconstruction frame stacking test to end up with exactly 1 machine frame.")
```
Comment (Gandalf2k15) [inline code/modules/unit_tests/rcd.dm]: why

**Comment by MLGTASTICa** [comment]
> 1. Your branch is hecked. Check files changed.
> 2. I'm fine with this existing but I'd rather you keep this in the armoury and not given to every officer. Up to three in the armoury is fine.
> 3. Please post a video of the breaching or post a .gif of it. Kinda need to see how this performs in-game.
> 4. Lower block chance significantly and make bullets not be blocked at all, or just have 2% block chance or something.
> 5. Consider making this something you wield as a requirement to smash the doors, it'll also make having high stamina damage okay if you are required to wield the hammer to use it effectively. Wielded can have higher block chance for melee attacks, considering you're holding a bar in front of you with two hands, as opposed to limply holding a huge hammer with one hand.
> 
> Overall very good PR, just needs tweaking.

Fixed the random file changes , gandalf said later to make sure to actually replace batongs so im confused on  the armoury. Video is here https://youtu.be/b3JdXguov_M ,  lowered block chance from 45% to 25% , 15%  > 5% . As for the last part , i'd actually add wielding for all items in another PR and apply it to the hammer later.

File: code/modules/unit_tests/rcd.dm (hunk 2 above)
Comment (MLGTASTICa) [reply code/modules/unit_tests/rcd.dm]: No idea why but i fixed it.

**Comment by PCPort** [comment]
> I have literally never seen a shove stun work on Skyrat. Shove fights are endless.
> 
> 15 seconds to hardstun a single person is frankly absurd. You're making the only option for security to even do their most basic job of arresting people for non-antag crimes an absurd chore or forcing them to just shoot everyone with rubbers (which will cause extreme problems with the crew)

As someone who has shovestunned many, *many* people on this server, I can attest that they do work.

**Comment by Ironnhawk** [comment]
This is something that should be in the armory, not standard issue.

**Comment by Gandalf2k15** [comment]
Fix the code issues and github issues eeeee

**Comment by Mithrun** [comment]
If this was in the armory as on optional tool, less of a weapon, this would be great. i'd love this. 

But as a weapon? especially as a main weapon? i really don't like this idea. 
First off, just imagining security officers swinging those big things around is pretty funny and ridiculous, makes me think of monster hunter, really.
On top of that, i think the nightstick is actually in a pretty good spot now, having the option to upgrade to a stunprod later on (though if it were me, i'd lock that behind research or have the upgrades in the armory). 

Overall, good idea but.. rough around the edges?

**Comment by Gandalf2k15** [comment]
After trialing this, it's best if these go in the armory, leave the officers with the normal batons.

**Comment by MLGTASTICa** [comment]
> After trialing this, it's best if these go in the armory, leave the officers with the normal batons.

This was originally meant to get rid of batons because they force knockdown , but eh , whatever.

**Comment by Gandalf2k15** [comment]
https://github.com/Skyrat-SS13/Skyrat-tg/pull/4695

**Comment by Gandalf2k15** [comment]
https://github.com/Skyrat-SS13/Skyrat-tg/pull/4695

File: html/changelog.html [hunk 3]
```diff
@@ -573,6 +574,7 @@ <h3 class="author">unit0016 updated:</h3>
 				<li class="bugfix">Interdyne can no longer steal station minerals.</li>
 			</ul>
 
+
```
Comment (Gandalf2k15) [inline html/changelog.html]:
```suggestion
```

File: html/changelog.html [hunk 4]
```diff
@@ -403,6 +403,7 @@ <h3 class="author">werewolfboyfriend updated:</h3>
 				<li class="rscadd">Interdyne has gotten a shipment of furniture and supplies in, and now has areas to sit and has restocked some old resources.</li>
 			</ul>
 
+
```
Comment (Gandalf2k15) [inline html/changelog.html]:
```suggestion
```

File: code/modules/unit_tests/rcd.dm [hunk 5]
```diff
@@ -44,3 +44,4 @@
 		frame_count++
 
 	TEST_ASSERT_EQUAL(frame_count, 1, "Expected no density machine deconstruction frame stacking test to end up with exactly 1 machine frame.")
+
```
Comment (Gandalf2k15) [inline code/modules/unit_tests/rcd.dm]:
```suggestion
```

Comment (Gandalf2k15) [inline code/modules/unit_tests/rcd.dm]:
```suggestion
```

**Pull request merged**